### PR TITLE
Added appId and appIdSuffix options to cli-hermes

### PR DESCRIPTION
**Pull request by BrendonSled**
Summary:
---------

Very simple change.

Added the appId and appIdSuffix from the platform-android package. This allows anyone with an app suffix like `.debug` to profile hermes.

Test Plan:
----------

This code is copy paste from the platform-android package.

### CODE REVIEW

File: packages/cli-hermes/src/profileHermes/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// @ts-ignore
```
Comment (thymikee) [inline packages/cli-hermes/src/profileHermes/index.ts:1]: please revert this

File: packages/cli-hermes/src/profileHermes/sourcemapUtils.ts [hunk 2]
```diff
@@ -1,11 +1,10 @@
 import {Config} from '@react-native-community/cli-types';
-import {logger, CLIError} from '@react-native-community/cli-tools';
+import {logger, CLIError,fetch} from '@react-native-community/cli-tools';
```
Comment (thymikee) [inline packages/cli-hermes/src/profileHermes/sourcemapUtils.ts:2]: need to run Prettier on that (`yarn lint --fix`)

File: packages/cli-hermes/src/profileHermes/downloadProfile.ts [hunk 3]
```diff
@@ -46,13 +46,18 @@ export async function downloadProfile(
   raw?: boolean,
   shouldGenerateSourcemap?: boolean,
   port?: string,
+  appId?: string,
+  appIdSuffix?: string,
```
Comment (thymikee) [inline packages/cli-hermes/src/profileHermes/downloadProfile.ts:50]: can you update the jsdoc comment for this function to be up-to-date?

**Review by thymikee — CHANGES_REQUESTED** [review]
Thanks! Why not :)

**Review by thymikee — APPROVED** [review]

**Comment by grabbou** [comment]
Thanks, could you provide a use case for this addition? 

At the same time, could you provide a description to our documentation? Simply open `md` file and add these options together with the description.

**Comment by AndrewMorsillo** [comment]
@grabbou An example of when this is necessary: my app uses react-native-firebase, in development mode it connects to a separate staging firebase project. To do that it needs to have a different android package name than the production app, so I use `--appIdSuffix=staging`. Without this PR I don't think I can use `profile-hermes` because it won't have the correct package name in development mode. I think this solves https://github.com/react-native-community/cli/issues/1438

**Comment by BrendonSled** [comment]
@grabbou this is a simple port of the same feature from the cli.
https://github.com/react-native-community/cli/blob/master/docs/commands.md#--appid-string

File: packages/cli-hermes/src/profileHermes/downloadProfile.ts [hunk 4]
```diff
@@ -69,7 +76,7 @@ export async function downloadProfile(
     // If --raw, pull the hermes profile to dstPath
     if (raw) {
       execSyncWithLog(
-        `adb shell run-as ${packageName} cat cache/${file} > ${dstPath}/${file}`,
+        `adb shell run-as ${packageNameWithSuffix} cat cache/${file} > ${dstPath}/${file}`,
```
Comment (grabbou) [inline packages/cli-hermes/src/profileHermes/downloadProfile.ts:79]: Do we need to run `as` with suffix? We're not doing this in other `adb` calls in this file.

**Review by grabbou — CHANGES_REQUESTED** [review]
Generally looks good, but I do have a question (left a comment).

File: packages/cli-hermes/src/profileHermes/downloadProfile.ts (hunk 4 above)
Comment (BrendonSled) [reply packages/cli-hermes/src/profileHermes/downloadProfile.ts:79]: That's a good question. I attempted to modify as little as possible. The prior command was `run-as` so I just used that

**Comment by BrendonSled** [comment]
> Thanks, could you provide a use case for this addition?
> 
> At the same time, could you provide a description to our documentation? Simply open `md` file and add these options together with the description.

For this I just copied this code from the cli 
https://github.com/react-native-community/cli/blob/641b21f583c97e3d48ce87d5fe804f42db92fa5c/packages/platform-android/src/commands/runAndroid/tryLaunchAppOnDevice.ts#L20

This allows app suffix support. Example: com.test**.development**, com.test**.staging**,

**Comment by BrendonSled** [comment]
Also @grabbou I would love to update the documentation for hermes. But I can't find the documentation for that specific cli.

The appidsuffix is already in the main cli commands, here:
https://github.com/react-native-community/cli/blob/master/docs/commands.md#--appidsuffix-string

Can you point me in the right direction?

**Comment by henrymoulton** [comment]
@BrendonSled have just come across this issue/PR as I'm needing to do a bit of Hermes profiling in an app that has Debug/Staging/Production flavors. 

@elizabeth-dev are you actively still needing this feature?

@thymikee  / @grabbou anything needed to get this merged in?

File: packages/cli-hermes/src/profileHermes/downloadProfile.ts (hunk 4 above)
Comment (grabbou) [reply packages/cli-hermes/src/profileHermes/downloadProfile.ts:79]: What I am saying is that on this line: https://github.com/react-native-community/cli/pull/1386/files#diff-13ca0a162291646faf5ecefc657d151b45ebf950563c9f32d38ec39ede1341b1R90 we are not using `packageNameWithSuffix`.

I guess as per your comment, that has to be modified.

**Comment by grabbou** [comment]
The PR is ready to be merged, I have applied some tweaks. @henrymoulton would you mind testing that everything works as expected for you?

**Comment by grabbou** [comment]
Meanwhile, I have decided to ship it. Since we're not going to make a release anytime soon and this PR is additive, I will continue working on the CLI and wait for reports in the meantime.

**Comment by joaobertacchi** [comment]
@grabbou, @BrendonSled  , I'm trying to use this new feature without success. Am I doing something wrong?
```
$ npx react-native@nightly profile-hermes --appId com.myapp ~/tmp
error: unknown option `--appId'
```

**Pull request merged**